src/main/java/com/_up/megastore/services/implementations/ProductService.java

```diff
@@ -85,4 +95,22 @@ public void ifImageURLExistsUpdateProductImageURL(MultipartFile multipartFile, P
             product.setImageURL(imageURL);
         }
     }
+
+    private void validateProductForRestoration(UUID productId) {
+        ifProductDoesNotExistThrowException(productId);
+        ifProductIsNotDeletedThrowException(productId);
+    }
+
+    private void ifProductDoesNotExistThrowException(UUID productId) {
+        if (!productRepository.existsById(productId)) {
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "Product with id " + productId + " does not exist.");
+        }
+    }
+
+    private void ifProductIsNotDeletedThrowException(UUID productId) {
+        if (!productRepository.existsByProductIdAndDeletedTrue(productId)) {
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "Product with id " + productId + " is not deleted.");
+        }
+    }
```

Comment:
Estamos haciendo dos peticiones a la BD de para practicarmente hacer lo mismo, podriamos hacer lo siguiente

1. Obtenerlo por ID y si no existe se lanza excepción
2. Si lo obtenemos, ver mediante un condicional si el producto está eliminado en caso de que no lo esté lanzamos excepción

```suggestion
    private void validateProductForRestoration(UUID productId) {
        Product product = productRepository.findById(productId).orElseThrow(() -> new ResponseStatusExceptionHttpStatus.BAD_REQUEST, "Product with id " + productId + " does not exist."));
        if (product.isDeleted == false) {
            throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "Product with id " + productId + " is not deleted.");
        } 
    }
```

y nos ahorramos una query a la BD

Reply:
Banco este cambio